File: top_k_frequent.rb
def top_k_frequent(nums, k)
  hash = Hash.new(0)
  nums.each do |el|
    hash[el] += 1
  end
  hash.keys.sort_by {|k| hash[k]}.reverse[0...k]
end

nums = [1,1,1,2,2,3]
p top_k_frequent(nums, 2)
# [1,2]

File: topKFrequent.js
var topKFrequent = function(nums, k) {
  let obj = {};
  for (let el of nums) {
    obj[el] = (obj[el] || 0) + 1;
  }
  let keys = Object.keys(obj);
  let sorted = keys.sort ((a,b) => {
    return obj[b] - obj[a];
  });
  return sorted.map (el => Number(el)).slice(0,k);
};

let num = [1,1,1,2,2,3];
let k = 2;
console.log(topKFrequent(num, k));
